### PR TITLE
Created a welcoming system for new members to Armada

### PR DESCRIPTION
**Pull request by mattpezzente**
Changed this from my forked repo, to a branch of the dev branch originating from reactivepixel's Max-Bot repo.

### CODE REVIEW

**Comment by brennondenny** [comment]
The bot works fine, looks like there is a conflict with the Airbnb ESLint, you can check that below.

**Comment by BHess2653** [comment]
Looks good to me everything passed and !welcome works.

File: bot/client.js [hunk 1]
```diff
@@ -15,6 +15,22 @@ client.on('ready', () => {
   util.log('Bot Online and Ready', 0);
 });
 
+// Actions for new guild members
+client.on('guildMemberAdd', (member) => {
+  // Send a the direct welcome message to a guild member
+  member.send(`Welcome ${member} to the Full Sail Armada! Please read our Terms of Service below to get started: \n\n --Terms of Service--`);
+  util.log('Direct Message Sent to New Member', 0);
```
Comment (reactivepixel) [inline bot/client.js]: Should this be a level 0 debug event?

But good use of including the utility's logging.

File: bot/client.js [hunk 2]
```diff
@@ -15,6 +15,22 @@ client.on('ready', () => {
...
+
+  // Variable to store channel if it exists
+  const channel = client.channels.find('name', 'general');
+  // Check if channel exists
+  if (channel) {
+    // Send a welcome broadcast tot he general channel
+    channel.send(`Welcome to the Full Sail Armada ${member}!`);
```
Comment (reactivepixel) [inline bot/client.js]: Proper sentence structure would be idea.

Welcome to the Full Sail Armada```,``` ${member}!

File: bot/client.js [hunk 3]
```diff
@@ -15,6 +15,22 @@ client.on('ready', () => {
   util.log('Bot Online and Ready', 0);
 });
 
+// Actions for new guild members
+client.on('guildMemberAdd', (member) => {
+  // Send a the direct welcome message to a guild member
+  member.send(`Welcome ${member} to the Full Sail Armada! Please read our Terms of Service below to get started: \n\n --Terms of Service--`);
```
Comment (reactivepixel) [inline bot/client.js]: Duplicate logic from your controller.

File: bot/controllers/welcome.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        this.welcomeAction.bind(controller),
+        'dm',
+      ),
+    ];
+  }
+
+  // Sends the welcome message to the user
+  welcomeAction() {
+    const { message } = this;
+    util.log('Sending Welcome Message to User', 0);
```
Comment (reactivepixel) [inline bot/controllers/welcome.js]: Should this be a level 0 debug event?

File: package.json [hunk 5]
```diff
@@ -29,7 +29,7 @@
     "mocha-eslint": "^4.1.0",
     "mysql2": "^1.4.2",
     "nodemailer": "^4.4.0",
-    "sequelize": "^4.22.5",
+    "sequelize": "^4.37.6",
```
Comment (reactivepixel) [inline package.json]: Why are you updating the version of Sequelize in a PR about a welcome controller. This should be a separate PR with requests for specific review of database interactions. Additionally, did you fully vet the impact of this update across the entire system's use of the database?

File: bot/controllers/welcome.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    // Aliasing 'this' as controller to allow for binding in actions
+    const controller = this;
+
+    // Array of all commands, see baseCommand.js for prototype
+    this.commands = [
+      new Command(
+        '!welcome',
+        '!welcome',
+        'Send Welcome Message',
+        'Sends A Direct Welcome Message.',
```
Comment (reactivepixel) [inline bot/controllers/welcome.js]: Sends a direct welcome message is slightly ambiguous, after all, it is sending "the" welcome message, not "a" welcome message. And more specifically, it is resending the welcome message to the initiating user.

**Review by reactivepixel — CHANGES_REQUESTED** [review]
Very good work overall @mattpezzente. Good commit messages and your PR is feature complete and is inline with a standard submission. Ideally I was looking for you to explore what it takes to invoke your controller within the guildAddUser event.

**Comment by mattpezzente** [comment]
Commit message typo: "util.js" -> "botUtils.js"

File: bot/client.js (hunk 1 above)
Comment (mattpezzente) [reply bot/client.js]: I believe so, as it is a log message that should appear on the production server, indicating an action that has taken place.

File: bot/controllers/welcome.js (hunk 4 above)
Comment (mattpezzente) [reply bot/controllers/welcome.js]: Changed this around and added it to a level 3 log, as I believe it has to do with an extended controller action, rather than an action close to the top level.

File: package.json (hunk 5 above)
Comment (mattpezzente) [reply package.json]: Mistake while re-installing packages, should be amended in latest commit.
